examples/analysis/point_extraction/interpolation.py

```diff
@@ -0,0 +1,56 @@
+"""
+Regular-grid interpolation
+==========================
+
+This example demonstrates raster interpolation to point values using :func:`~geoutils.Raster.interp_points`.
+"""
+# %%
+# We open an example raster, a digital elevation model in South America
+
+# sphinx_gallery_thumbnail_number = 2
+import geoutils as gu
+
+rast = gu.Raster(gu.examples.get_path("exploradores_aster_dem"))
+rast.crop([rast.bounds.left, rast.bounds.bottom, rast.bounds.left + 2000, rast.bounds.bottom + 2000])
+
+# Plot the raster
+rast.show(cmap="terrain")
+
+# %%
+# We generate a random subsample of 100 points to interpolate, and extract the coordinates
+
+import geopandas as gpd
+import numpy as np
+
+# Replace by Raster function once done (valid coords)
+np.random.seed(42)
+x_coords = np.random.uniform(rast.bounds.left + 50, rast.bounds.right - 50, 50)
+y_coords = np.random.uniform(rast.bounds.bottom + 50, rast.bounds.top - 50, 50)
+
+vals = rast.interp_points(pts=list(zip(x_coords, y_coords)))
+
+# %%
+# Replace by Vector function once done
+ds = gpd.GeoDataFrame(geometry=gpd.points_from_xy(x=x_coords, y=y_coords), crs=rast.crs)
+ds["vals"] = vals
+ds.plot(column="vals", cmap="terrain", legend=True, vmin=np.nanmin(rast), vmax=np.nanmax(rast), marker="x")
+
+# %%
+# .. important::
+#       The interpretation of where raster values are located differ, see XXX. The parameter ``shift_area_or_point`` (off by default) can be turned on to ensure
```

Comment:
Do you plan to include a section about this in the doc?

Reply:
Yes, added in #361